### PR TITLE
As/mega code review evm 474

### PR DESCRIPTION
**Pull request by alsala**

### CODE REVIEW

**Review by saratnt — APPROVED** [review]

File: sdk/src/main/scala/com/horizen/account/utils/MainchainTxCrosschainOutputAddressUtil.scala [hunk 1]
```diff
@@ -8,7 +8,7 @@ object MainchainTxCrosschainOutputAddressUtil {
   // MC prepends a padding of "0 bytes" (if needed) in the ccout address up to the length of 32 bytes.
   // After reversing the bytes, the padding is trailed to the correct 20 bytes proposition
   def getAccountAddress(inputAddress: Array[Byte]): Array[Byte] = {
-    require(inputAddress.length == 32)
+    require(inputAddress.length == 32, s"byte array length ${inputAddress.length} != 32")
```
[the remark this answers is not in the document]
Comment (alsala) [reply sdk/src/main/scala/com/horizen/account/utils/MainchainTxCrosschainOutputAddressUtil.scala]: Ok, done

[the remark this answers is not in the document]
Comment (alsala) [reply sdk/src/main/scala/com/horizen/account/utils/MainchainTxCrosschainOutputAddressUtil.scala]: Actually it is not (Account.ADDRESS_SIZE =20), the check should be done against 32 byte length

[the remark this answers is not in the document]
Comment (saratnt) [reply sdk/src/main/scala/com/horizen/account/utils/MainchainTxCrosschainOutputAddressUtil.scala:11]: OK

File: sdk/src/main/scala/com/horizen/account/certificatesubmitter/AccountCertificateSubmitter.scala [hunk 2]
```diff
@@ -46,6 +47,13 @@ class AccountCertificateSubmitter[T <: CertificateData](settings: SidechainSetti
   type MP = AccountMemoryPool
   type PM = AccountBlock
 
+  //override def getUtxoMerkleTreeRoot(referencedEpoch: Int, state: AccountState): Optional[Array[Byte]] = Optional.empty()
+
+  /*
+  override def getWithdrawalRequests(state: AccountState, referencedEpochNumber: Int): Seq[BackwardTransfer] =
+    state.getWithdrawalRequests(referencedEpochNumber).map(request => new BackwardTransfer(request.proposition.bytes, request.valueInZennies))
+  */
+
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/account/certificatesubmitter/AccountCertificateSubmitter.scala:56]: changes in this file can be reverted back, seems to be a drawback of the merge

File: sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala [hunk 3]
```diff
@@ -408,10 +408,12 @@ class ForgerStakeMsgProcessorTest
     }
   }
 
+  // this test is not meaningful anymore: all sanity checks are performed before calling any MessageProcessor
+  // ===> here we are succesful even with an empty balance account
   @Test
+  @Ignore
```
Comment (i-Alex) [inline sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala:414]: do we need to keep this ignored test at all?

File: sdk/src/main/scala/com/horizen/account/certificatesubmitter/AccountCertificateSubmitter.scala (hunk 2 above)
Comment (alsala) [reply sdk/src/main/scala/com/horizen/account/certificatesubmitter/AccountCertificateSubmitter.scala]: Ok, done

File: sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala (hunk 3 above)
Comment (alsala) [reply sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala]: No you are right, I removed it

**Review by i-Alex — APPROVED** [review]

**Pull request merged**